src/light-godwoken/__tests__/lightGodwokenDeposit.spec.ts

```diff
@@ -0,0 +1,192 @@
+import sinon from "sinon";
+import LightGodwokenV1 from "../LightGodwokenV1";
+import LightGodwokenV0 from "../LightGodwokenV0";
+import DefaultLightGodwokenProvider from "../lightGodwokenProvider";
+import { generateCellInput, outputCapacityOf, randomScript, outputSudtAmountOf } from "./utils";
+
+let lightGodwokenV0: LightGodwokenV0;
+let lightGodwokenV1: LightGodwokenV1;
+let lightGodwokenProviderV0: DefaultLightGodwokenProvider;
+let lightGodwokenProviderV1: DefaultLightGodwokenProvider;
+beforeEach(() => {
+  const ethAddress = "0x0C1EfCCa2Bcb65A532274f3eF24c044EF4ab6D73";
+  const dummyEthereum = {
+    on: () => {},
+  };
+  lightGodwokenProviderV1 = new DefaultLightGodwokenProvider(ethAddress, dummyEthereum, "v1");
+  lightGodwokenV1 = new LightGodwokenV1(lightGodwokenProviderV1);
+  lightGodwokenProviderV0 = new DefaultLightGodwokenProvider(ethAddress, dummyEthereum, "v0");
+  lightGodwokenV0 = new LightGodwokenV0(lightGodwokenProviderV0);
+});
+
+describe("test light godwoken v1 deposit", () => {
+  it("should generateDepositLock works fine", async () => {
+    const lock = await lightGodwokenV1.generateDepositLock();
+    expect(lock).toEqual({
+      args: "0x4940246f168f4106429dc641add3381a44b5eef61e7754142f594e986671a575a10000001000000030000000990000003837aad0e28da55d366d62b7df9b1b0613c39c730c4c409b9722d4bed8cfa9266900000010000000300000003100000010571f91073fdc3cdef4ddad96b4204dd30d6355f3dda9a6d7fc0fa0326408da01340000004940246f168f4106429dc641add3381a44b5eef61e7754142f594e986671a5750c1efcca2bcb65a532274f3ef24c044ef4ab6d73b0040000000000c0",
+      code_hash: "0xcc2b4e14d7dfeb1e72f7708ac2d7f636ae222b003bac6bccfcf8f4dfebd9c714",
+      hash_type: "type",
+    });
+  });
+  it("should deposit 2000 ckb ok when user balance is 2000", async () => {
+    const mockCell = generateCellInput(2000);
+
+    const mockCollector = { collect: sinon.stub().onFirstCall().returns([mockCell]) };
+    sinon.stub(lightGodwokenProviderV1.ckbIndexer, "collector").returns(mockCollector);
```

Comment:
test like this is hard to write, It's because we can not get our query param in `Collector.collector`. so we need to use call count to ensure function return

I think we can move stub to `beforeAll`hook,and do stub like below
```suggestion
    const mockCollector = { collect: sinon.stub().onFirstCall().returns([mockCell]) };
    sinon.stub(lightGodwokenProviderV1.ckbIndexer, "collector").returns((query) => {
      if(query is ckb query) {
         return ckbCollector
      }
      
       if(query is sudt query) {
         return sudtCollector
      }
    });
```